### PR TITLE
Clean up icon related theme code

### PR DESCRIPTION
**Pull request by fmoc**
Contains some of #10606, excluding the overlay part (which is planned for [5.0](https://github.com/owncloud/client/milestone/75)).

### CODE REVIEW

File: src/libsync/theme.cpp [hunk 1]
```diff
@@ -466,64 +466,62 @@ bool Theme::aboutShowCopyright() const
...
+            Q_FALLTHROUGH();
+        case SyncResult::Paused:
+            statusIcon = QStringLiteral("state-pause");
+            break;
+        case SyncResult::SyncPrepare:
+            Q_FALLTHROUGH();
+        case SyncResult::Success:
+            if (result.hasUnresolvedConflicts()) {
+                return syncStateIconName(SyncResult{SyncResult::Problem}, sysTray, sysTrayMenuVisible);
+            }
```
Comment (TheOneRing) [inline src/libsync/theme.cpp:490]: if you move the offline case we could use a fallthrough instead of the recursive call, which might make sings clearer.

**Review by TheOneRing — APPROVED** [review]

File: src/libsync/theme.cpp [hunk 2]
```diff
@@ -466,64 +466,61 @@ bool Theme::aboutShowCopyright() const
...
+            Q_FALLTHROUGH();
+        case SyncResult::SyncRunning:
+            statusIcon = QStringLiteral("state-sync");
+            break;
+        case SyncResult::SyncAbortRequested:
+            Q_FALLTHROUGH();
+        case SyncResult::Paused:
+            statusIcon = QStringLiteral("state-pause");
+            break;
+        case SyncResult::SyncPrepare:
```
Comment (TheOneRing) [inline src/libsync/theme.cpp:485]: the fix up is wrong, you need to move offline else, SyncPrepare will now get the offline icon

**Review by TheOneRing — APPROVED** [review]

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14753/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/14753/oc10/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/14753/oc10/guiReportUpload/serverlog.log

Screenshots:
- [Syncing_a_file_from_the_server_and_creating_a_conflict.png](https://cache.owncloud.com/public/owncloud/client/14753/oc10/guiReportUpload/screenshots/Syncing_a_file_from_the_server_and_creating_a_conflict.png)
- [sharing_file_folder_with_a_user_that_has_special_characters_as_username.png](https://cache.owncloud.com/public/owncloud/client/14753/oc10/guiReportUpload/screenshots/sharing_file_folder_with_a_user_that_has_special_characters_as_username.png)

</details>

**Comment by HanaGemela** [comment]
Passed in ownCloud 4.0.0.10542-daily20230331, mac

**Pull request merged**